src/Serializer/AbstractItemNormalizer.php

```diff
@@ -122,14 +134,33 @@ protected function extractAttributes($object, $format = null, array $context = [
     /**
      * {@inheritdoc}
      */
-    protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
-    {
+    protected function getAllowedAttributes(
+        $classOrObject,
+        array $context,
+        $attributesAsString = false
+    ) {
         $options = $this->getFactoryOptions($context);
-        $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
+        $propertyNames = $this
+            ->propertyNameCollectionFactory
+            ->create($context['resource_class'], $options);
 
         $allowedAttributes = [];
         foreach ($propertyNames as $propertyName) {
-            $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $propertyName, $options);
+            $propertyMetadata = $this
+                ->propertyMetadataFactory
+                ->create($context['resource_class'], $propertyName, $options);
+
+            if (
+                isset($context['api_denormalize'])
+                    && !$propertyMetadata->isWritable()
+                    && !$propertyMetadata->isIdentifier()
+            ) {
+                throw new RuntimeException(sprintf(
+                    'Property \'%s.%s\' is not writeable',
+                    $context['resource_class'],
+                    $propertyName
+                ));
+            }
```

Comment:
Yeah. I actually a made an exception in case the field is an identifier, as quick way out. I agree it deserves some discussion, but it took me soooooo long to figure this out I thought it might be helpful (: